api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1138,7 +1138,12 @@ private HttpResponse executeReq(Method method, String region, String bucketName,
           errorXml = scanner.next();
         }
 
-        errorResponse = new ErrorResponse(new StringReader(errorXml));
+        if ("application/xml".equals(response.headers().get("content-type"))) {
+          errorResponse = new ErrorResponse(new StringReader(errorXml));
+        } else {
+          errorResponse = new ErrorResponse("Non-Xml Response Received");
```

Comment:
This is incorrect usage or addition. Each class and error has its own purpose. You would need a new exception `InvalidResponseException` like `NoResponseException`.

Go through javadoc http://minio.github.io/minio-java/ completely.